### PR TITLE
[plugin.video.cbc] 4.0.0

### PR DESCRIPTION
**Pull request by micahg**
### Description
A working CBC plugin for the new API, including: live TV as well as live and on-demand programming.
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by micahg** [comment]
Hello,

Just a preemptive message about submitting a PR against an add-on for which I am not the maintainer. I have @learningit's permission to take this over (discussed at https://forum.kodi.tv/showthread.php?tid=328421&pid=2703229#pid2703229).

Please let me know what changes you would like before merging.

Thanks in advance.

**Comment by enen92** [comment]
@micahg  please rename the PR in the formar "[plugin] version"

File: plugin.video.cbc/addon.xml [hunk 1]
```diff
@@ -1,27 +1,24 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <addon id="plugin.video.cbc"
        name="Canadian Broadcasting Corp (CBC)"
-       version="3.0.3"
-       provider-name="t1m">
+       version="4.0.0"
+       provider-name="micahg">
```
Comment (enen92) [inline plugin.video.cbc/addon.xml]: I don't think this is fair, I'd prefer if you add yourself as an author along @learningit. But it is really up to him.

File: plugin.video.cbc/addon.xml [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
-            library="default.py">
-        <provides>video</provides>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+	<provides>video</provides>
   </extension>
   <extension point="xbmc.addon.metadata">
-    <summary lang="en">Canadian Broadcasting Corp (CBC)</summary>
-    <description lang="en">CBC Videos - only for use in Canada. CBC.ca is Canada's Online Information Source. Comprehensive web site for news, entertainment, sports, and business.</description>
-    <disclaimer lang="en">Feel free to use this script. For information visit the wiki.</disclaimer>
+    <summary>CBC TV Kodi Plugin</summary>
```
Comment (enen92) [inline plugin.video.cbc/addon.xml]: missing language tags

File: plugin.video.cbc/addon.xml [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
+    <description>View online content from the Canadian Broadcasting Corporation.</description>
```
Comment (enen92) [inline plugin.video.cbc/addon.xml]: missing lang tag

File: plugin.video.cbc/addon.xml [hunk 4]
```diff
@@ -1,27 +1,24 @@
...
     <platform>all</platform>
     <language>en</language>
+    <disclaimer>For bugs, requests or general questions visit the CBC TV thread on the Kodi forum.</disclaimer>
```
Comment (enen92) [inline plugin.video.cbc/addon.xml]: missing lang tag

File: plugin.video.cbc/default.py [hunk 5]
```diff
@@ -1,13 +1,190 @@
...
+        labels[key] = labels[key][0]
+    shows = Shows()
+    res = shows.getStream(smil)
+    item = xbmcgui.ListItem(labels['title'])
+
+    item.setInfo(type="Video", infoLabels=labels)
+    item.setIconImage(image)
+    p = xbmc.Player()
+    p.play(res['url'], item)
+    return
```
Comment (enen92) [inline plugin.video.cbc/default.py]: you don't need return

File: plugin.video.cbc/default.py [hunk 6]
```diff
@@ -1,13 +1,190 @@
...
+
+def playSmil(smil, labels, image):
+    cbc = CBC()
+    url = cbc.parseSmil(smil)
+    item = xbmcgui.ListItem(labels['title'])
+    item.setIconImage(image)
+    item.setInfo(type="Video", infoLabels=labels)
+    p = xbmc.Player()
+    p.play(url, item)
+    return
```
Comment (enen92) [inline plugin.video.cbc/default.py]: you don't need the return

File: plugin.video.cbc/default.py [hunk 7]
```diff
@@ -1,13 +1,190 @@
...
+            'labels': urlencode(labels),
+            'image': image
+        }
+        xbmcplugin.addDirectoryItem(handle=addon_handle,
+                                    url=sys.argv[0] + "?" + urlencode(values),
+                                    listitem=item,
+                                    isFolder=True)
+
+    xbmcplugin.endOfDirectory(addon_handle)
+    return
```
Comment (enen92) [inline plugin.video.cbc/default.py]: same

File: plugin.video.cbc/default.py [hunk 8]
```diff
@@ -1,13 +1,190 @@
...
+            values['labels'] = urlencode(labels)
+
+        plugin_url = sys.argv[0] + "?" + urlencode(values)
+        xbmcplugin.addDirectoryItem(handle=addon_handle,
+                                    url=plugin_url,
+                                    listitem=item,
+                                    isFolder = True)
+
+    xbmcplugin.endOfDirectory(addon_handle)
+    return
```
Comment (enen92) [inline plugin.video.cbc/default.py]: same

File: plugin.video.cbc/default.py [hunk 9]
```diff
@@ -1,13 +1,190 @@
...
+        labels = { 'title': menu_item }
+        item = xbmcgui.ListItem(menu_item)
+        item.setInfo(type="Video", infoLabels=labels)
+        values = { 'menu': menu_item }
+        xbmcplugin.addDirectoryItem(handle=addon_handle,
+                                    url=sys.argv[0] + "?" + urlencode(values),
+                                    listitem=item,
+                                    isFolder=True)
+    xbmcplugin.endOfDirectory(addon_handle)
+    return
```
Comment (enen92) [inline plugin.video.cbc/default.py]: same

File: plugin.video.cbc/resources/lib/cbc.py [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+        session_cookies = loadCookies()
+        if not session_cookies == None: 
+            self.session.cookies = session_cookies 
+        return
+
+
+    def getRegistrationURL(self):
+        r = self.session.get(self.IDENTITIES_URL)
+        if not r.status_code == 200:
+            log('ERROR: {} returns status of {}'.format(url, r.status_code), True)
```
[the remark this answers is not in the document]
Comment (enen92) [reply plugin.video.cbc/resources/lib/cbc.py]: ^^

File: plugin.video.cbc/addon.xml (hunk 1 above)
Comment (learningit) [reply plugin.video.cbc/addon.xml]: I have no issues at all with this. I suggested that he take over this I'd as I no longer maintain the add-on.

Comment (learningit) [reply plugin.video.cbc/addon.xml]: ID not I'd in above

Comment (learningit) [reply plugin.video.cbc/addon.xml]: ID not I'd in above

Comment (enen92) [reply plugin.video.cbc/addon.xml]: fine to me then

File: plugin.video.cbc/resources/lib/shows.py [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+        status = dom.getElementsByTagName('status')[0].firstChild.nodeValue
+        if not status == 'success':
+            log('ERROR: {} has status of {}'.format(url, status), True)
+            return None
+
+        return {
+            'url': dom.getElementsByTagName('url')[0].firstChild.nodeValue
+        }
+
+        return None
```
[the remark this answers is not in the document]
Comment (enen92) [reply plugin.video.cbc/resources/lib/shows.py]: ^^

File: plugin.video.cbc/addon.xml (hunk 1 above)
Comment (micahg) [reply plugin.video.cbc/addon.xml]: Actually, as soon as I submitted this I thought its probably too much also. What is the syntax for sharing - comma-delimiting?

File: plugin.video.cbc/default.py (hunk 5 above)
Comment (micahg) [reply plugin.video.cbc/default.py]: Force of habit -- i'll clean them up.

File: plugin.video.cbc/addon.xml (hunk 1 above)
Comment (micahg) [reply plugin.video.cbc/addon.xml]: NVM, just read https://kodi.wiki/view/Addon.xml#provider-name_attribute

**Comment by micahg** [comment]
Hello,

Requested changes applied.

Thanks!

**Comment by Rechi** [comment]
Can also give the commit the correct message `[plugin.video.cbc] 4.0.0` and adjust the PR title too?

**Comment by micahg** [comment]
Yeah, whoops.

**Comment by enen92** [comment]
Thanks for adjusting

**Comment by micahg** [comment]
NP, tyvm.

**Pull request merged**